pip/__init__.py

```diff
@@ -181,19 +215,44 @@ def parseopts(args):
     # the subcommand name
     cmd_name = args_else[0]
 
-    if cmd_name not in commands_dict:
-        guess = get_similar_commands(cmd_name)
-
-        msg = ['unknown command "%s"' % cmd_name]
-        if guess:
-            msg.append('maybe you meant "%s"' % guess)
-
-        raise CommandError(' - '.join(msg))
-
     # all the args without the subcommand
     cmd_args = args[:]
     cmd_args.remove(cmd_name)
 
+    # Autocorrect command name
+    if cmd_name not in commands_dict:
+        vals = _get_autocorrect_configuration_values(parser.config)
+        wait_time, suggest_cut_off, replace_cut_off = vals
+
+        score, guess = get_similar_command(cmd_name, suggest_cut_off)
+
+        if guess is None:  # nothing similar
+            err_msg = 'pip does not have a command "%s"' % cmd_name
+        elif score < replace_cut_off:  # not similar enough
+            err_msg = (
+                'pip does not have a command "%s" - did you mean "%s"?'
+            ) % (cmd_name, guess)
+        else:  # close enough - no error
+            err_msg = None
+
+        if err_msg is not None:
+            raise CommandError(err_msg)
+
+        msg = (
+            'You called a pip command named "%s" which does not exist.\n'
+            'Assuming you meant "%s", pip will continue in %.1f seconds...'
+        )
+
+        logger.warning(msg, cmd_name, guess, wait_time)
+        try:
+            time.sleep(wait_time)
+        except KeyboardInterrupt:
+            logger.critical('Operation cancelled by user')
+            logger.debug('Exception information:', exc_info=True)
```

Comment:
is it really a good idea to just go with the autocorrected command?
this may hide severely broken scripts and problems for longer than necessary,

imho the tool should still fail